File: easy/python/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = dict()
        for i in range(len(nums)):
            element_to_find = target - nums[i]
            if element_to_find in hashmap:
                return [i, hashmap[element_to_find]]
            hashmap[nums[i]] = i

File: algorithms/easy/golang/1-two-sum.go
package golang

func twoSum(nums []int, target int) []int {
	discoveredNums := make(map[int]int)
	for i, num := range nums {
		if j, ok := discoveredNums[target-num]; ok {
			return []int{j, i}
		}
		discoveredNums[num] = i
	}
	return []int{-1, -1}
}
